### PR TITLE
Fix failing test for Numba Func

### PR DESCRIPTION
**Pull request by majra20**
#### Why we need this PR?
- It fixes a bug: Data passed in test_operator_numba_func:test_multiple_ins has wrong shape. It shows that we should make additional checks if user populated data correctly.

#### What happened in this PR?
 - What solution was applied:
     *[ Passed correct data to test and add checks if data passed by user is in same shape that user expects ]*
 - Affected modules and functionalities:
     *[ Additional checks in numba_func.cc ]*
 - Key points relevant for the review:
     *[ NA ]*
 - Validation and testing:
     *[ NA ]*
 - Documentation (including examples):
     *[ NA ]*

### CODE REVIEW

File: dali/operators/numba_function/numba_func.cc [hunk 1]
```diff
@@ -173,46 +173,61 @@ NumbaFuncImpl<Backend>::NumbaFuncImpl(const OpSpec &spec) : Base(spec) {
...
-  in_shapes_.resize(in_types_.size());
-  for (size_t in_id = 0; in_id < in_types_.size(); in_id++) {
-    in_shapes_[in_id] = ws.InputRef<CPUBackend>(in_id).shape();
+  in_shapes_.resize(ninputs);
+  for (int in_id = 0; in_id < ninputs; in_id++) {
+    auto& in = ws.InputRef<CPUBackend>(in_id);
+    in_shapes_[in_id] = in.shape();
+    DALI_ENFORCE(in_shapes_[in_id].sample_dim() == ins_ndim_[in_id], make_string(
+      "Dimension passed in `ins_ndim` at index ", in_id,
+      " doesn't match dimension of input data."));
```
Comment (jantonguirao) [inline dali/operators/numba_function/numba_func.cc:190]:
```suggestion
    DALI_ENFORCE(in_shapes_[in_id].sample_dim() == ins_ndim_[in_id], make_string(
      "Number of dimensions passed in `ins_ndim` at index ", in_id,
      " doesn't match the number of dimensions of the input data: ", in_shapes_[in_id].sample_dim(), " != ", ins_ndim_[in_id]));
```

File: dali/operators/numba_function/numba_func.cc [hunk 2]
```diff
@@ -173,46 +173,61 @@ NumbaFuncImpl<Backend>::NumbaFuncImpl(const OpSpec &spec) : Base(spec) {
...
+    DALI_ENFORCE(in.type().id() == in_types_[in_id], make_string(
+      "Type passed in `in_types` at index ", in_id, " doesn't match type of input data."));
```
Comment (jantonguirao) [inline dali/operators/numba_function/numba_func.cc:192]:
```suggestion
    DALI_ENFORCE(in.type().id() == in_types_[in_id], make_string(
      "Data type passed in `in_types` at index ", in_id, " doesn't match type of the input data: ", in.type().id(), " != ", in_types_[in_id]));
```

**Review by jantonguirao — APPROVED** [review]

**Comment by majra20** [comment]
!build

File: dali/operators/numba_function/numba_func.cc [hunk 3]
```diff
@@ -173,46 +173,63 @@ NumbaFuncImpl<Backend>::NumbaFuncImpl(const OpSpec &spec) : Base(spec) {
...
-  in_shapes_.resize(in_types_.size());
-  for (size_t in_id = 0; in_id < in_types_.size(); in_id++) {
-    in_shapes_[in_id] = ws.InputRef<CPUBackend>(in_id).shape();
+  in_shapes_.resize(ninputs);
+  for (int in_id = 0; in_id < ninputs; in_id++) {
+    auto& in = ws.InputRef<CPUBackend>(in_id);
+    in_shapes_[in_id] = in.shape();
+    DALI_ENFORCE(in_shapes_[in_id].sample_dim() == ins_ndim_[in_id], make_string(
+      "Number of dimensions passed in `ins_ndim` at index ", in_id,
+      " the number of dimensions of the input data: ",
```
Comment (JanuszL) [inline dali/operators/numba_function/numba_func.cc:190]:
```suggestion
      " is: ",
```

File: dali/test/python/test_operator_numba_func.py [hunk 4]
```diff
@@ -199,11 +199,8 @@ def numba_multiple_ins_pipe(shapes, dtype, run_fn=None, out_types=None, in_types
     data2 = fn.external_source(lambda: get_data_zeros(shapes, dtype), batch=True, device = "cpu")
     return numba_func(data0, data1, data2, run_fn=run_fn, out_types=out_types, in_types=in_types, outs_ndim=outs_ndim, ins_ndim=ins_ndim, setup_fn=setup_fn, batch_processing=batch_processing)
 
-    out = numba_func(images_in, run_fn=run_fn, out_types=out_types, in_types=in_types, outs_ndim=outs_ndim, ins_ndim=ins_ndim, setup_fn=setup_fn, batch_processing=batch_processing)
-    return images_in, out0, out1, out2
-
 def test_multiple_ins():
-    pipe = numba_multiple_ins_pipe(shapes=(10, 10), dtype=np.uint8, batch_size=8, num_threads=1, device_id=0, run_fn=multiple_ins_run, setup_fn=multiple_ins_setup,
+    pipe = numba_multiple_ins_pipe(shapes=[(10, 10)], dtype=np.uint8, batch_size=8, num_threads=1, device_id=0, run_fn=multiple_ins_run, setup_fn=multiple_ins_setup,
```
Comment (JanuszL) [inline dali/test/python/test_operator_numba_func.py:203]: Is this the fix?

Comment (majra20) [reply dali/test/python/test_operator_numba_func.py:203]: Yes, that fixes test.

**Review by JanuszL — APPROVED** [review]

**Comment by majra20** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**